src/stream/src/executor/sort.rs

```diff
@@ -123,13 +123,17 @@ impl<S: StateStore> SortExecutor<S> {
         for msg in input {
             match msg? {
                 Message::Watermark(watermark) => {
-                    let Watermark { col_idx, val } = watermark.clone();
+                    let Watermark {
+                        col_idx,
+                        val,
+                        data_type: _,
+                    } = &watermark;
```

Comment:
Why ref here? It seems that the later `val.clone` can be avoided

Reply:
We'll finally yield it, so a `clone` is unavoidable. 🤔